src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/CoreCLRReadyToRunTypesTableNode.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+using Internal.NativeFormat;
+using Internal.Runtime;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    public class CoreCLRReadyToRunTypesTableNode : ObjectNode, ISymbolDefinitionNode
+    {
+        TargetDetails _target;
+        
+        List<(int Rid, EETypeNode Node)> _eeTypeNodes;
+        
+        public CoreCLRReadyToRunTypesTableNode(TargetDetails target)
+        {
+            _target = target;
+            _eeTypeNodes = new List<(int Rid, EETypeNode Node)>();
+        }
+        
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix);
+            sb.Append("__ReadyToRunAvailableTypesTable");
+        }
+
+        public int Offset => 0;
+
+        public override bool IsShareable => false;
+
+        protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public override ObjectNodeSection Section
+        {
+            get
+            {
+                if (_target.IsWindows)
+                    return ObjectNodeSection.ReadOnlyDataSection;
+                else
+                    return ObjectNodeSection.DataSection;
+            }
+        }
+
+        public int Add(int rid, EETypeNode eeTypeNode)
```

Comment:
There's no node type representing a CoreCLR type right now, but I don't think you'll want to have the CoreRT EETypeNode anywhere near the dependency graph for a CoreCLR ready to run compilation.

Reply:
I know we shouldn't emit the EE types themselves into the R2R executable - do you have any suggestions on what is the proper way to populate the R2R type table?

Reply:
`EETypeNode` has a bunch of logic that inserts various dependencies into the graph that the EEType data structure requires. The CoreCLR ready to run compilation will want its own node kind that doesn't do any of the EEType specific things (but potentially does some CoreCLR specific things, if needed).

This code is fine as it is now, but once someone starts hitting the problems with the dependency graph, I want to make sure there's enough police tape around this code to make it obvious that the fix is _not_ to update EETypeNode.